verify.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+set -e
+
+function validate() {
+    python_version=$1
+
+    echo "Validating version $python_version..."
+
+    tag=py-ipfs-http-client-verify:$python_version
+
+    docker build --build-arg PYTHON_VERSION=$python_version -t $tag .
+    docker run -it -v $(pwd):/source:ro -w /usr/src/app $tag tox -e styleck -e typeck
```

Comment:
Use `docker run --rm` to ensure the used image doesn't stick around needlessly.

Reply:
> You just dropped all the Docker-related code I tried to review. What is the plan there?

I thought I was getting some push back on it so I decided to pare down the PR to changes to the library. I can add it back if you like, with changes per the comments you added.

FYI - the linting step of the Travis build now passes. I don't have fixes for the remaining type checks since I'm not familiar enough with protocols to address it.

Do you want me to add the Docker scripts back into this PR, or open a new PR for it?

Reply:
Please open a new PR for the Docker stuff. The current PR is indeed useful as is, so no need to clutter it further.